site/source/docs/porting/simd.rst

```diff
@@ -1,52 +1,39 @@
-==============================
+=======================================
+Porting SIMD code targeting WebAssembly
+=======================================
 
-Emscripten utilizes the `SIMD.js specification <https://tc39.github.io/ecmascript_simd/>`_ to enable support for building SIMD code. While the specification does not expose direct access to native CPU SIMD instructions, it does offer the most basic operations that codebases commonly need. Given the nature of the SIMD.js specification, compiled SIMD code is platform-abstract and the generated .js files will run accelerated on both x86 SSE and ARM NEON enabled CPUs (and any other vector instruction sets that browsers might be targeting), although performance can vary.
+Emscripten supports the `WebAssembly SIMD proposal <https://github.com/webassembly/simd/>`_ when using the WebAssembly LLVM backend. To enable SIMD, pass the -msimd128 flag at compile time. This will also turn on LLVM's autovectorization passes, so no source modifications are necessary to benefit from SIMD.
 
-For backwards compatibility, all generated SIMD code will embed a polyfill for all SIMD.js instructions. This allows Emscripten compiled pages to be run even on browsers that don't have SIMD support available. However, due to the nature of the interoperation with asm.js, the SIMD.js polyfill will likely be much slower than scalar execution, so it is best to primarily target browsers that do support SIMD.js in hardware.
+At the source level, the GCC/Clang `SIMD Vector Extensions <https://gcc.gnu.org/onlinedocs/gcc/Vector-Extensions.html>`_ can be used and will be lowered to WebAssembly SIMD instructions where possible. A portable intrinsics header for WebAssembly SIMD is also being actively developed.
 
-Enabling SIMD code generation
-=============================
+WebAssembly SIMD is not supported when using the Fastcomp backend.
 
-There are three different ways to generate code to benefit from SIMD instructions:
+==================================
+Porting SIMD code targeting asm.js
+==================================
 
-- The easiest way is to utilize LLVM autovectorization support. This does not require any code changes and it is the most straightforward option to try for codebases that have not explicitly been written to target SIMD. To enable autovectorization, pass the linker flag -s SIMD=1 both when compiling and linking output files. Note that not all code can be autovectorized in this manner by the compiler, so the performance gains will greatly depend on the structure of the code in its hot paths.
+Emscripten utilizes a polyfill of the `SIMD.js specification <https://tc39.github.io/ecmascript_simd/>`_ to enable support for building SIMD code. Note that no modern JS engine supports SIMD.js natively, so using Emscripten's SIMD.js support will almost always be slower than building without SIMD. As such, Emscripten's SIMD.js support should be considered deprecated and may be removed in a future release.
```

Comment:
do we still use the polyfill?

Reply:
There must be a polyfill somewhere, and I assumed we still supplied it. I haven't removed it yet, if that's what you mean. I started trying to remove it but it was too pervasive so I only removed the SSE support.